### ISSUE TITLE
Exploring the Solar System with React, Node and the NASA APIs (Tim Troup)

### ISSUE THREAD

**Issue by timtroup** [techmeetup/edinburgh#132]
As a STEM “Ambassador” I volunteer time and support to promote STEM subjects to young learners

http://www.stemnet.org.uk/  

In this talk we shall explore how open data and open APIs can be used to create opportunities to inspire young people in STEM. In particular I will describe a single page web application I have built using React and the latest ES2016/ES2017 features that consumes open NASA APIs (https://api.nasa.gov/api.html) to create a Mars Rover photo gallery. An initial prototype of which can be found at:

http://somerledsolutions.com/mars-rover/

It almost seems churlish to complain that the majority of the photos that have made the 141 million mile journey across the solar system are mostly black and white. However the addition of colour can really make these photos far more engaging and easier to relate to. I will briefly introduce a ground breaking piece of recent research which has delivered software that can automatically colorize grayscale images that combines both global priors and local image features: 

http://hi.cs.waseda.ac.jp/~iizuka/projects/colorization/en/

I will demonstrate how to wrap this command line driven application and expose it as a RESTful service via Node. Integrating with the photo gallery application via a simple API call using the HTML 5 fetch standard. Let there be colour!

In summary we will boldly go where few JavaScript programs have gone before. I hope you will join me.

**Comment by marksutherland** [comment]
Hi @timtroup ! Sorry for the late response, this slipped past me until now.

This talk sounds really interesting and we'd be happy to have you. Have you been to TMU before?

We have space in May (10th) and a June (14th) if either of those suit you.

Thanks for volunteering!

**Comment by timtroup** [comment]
Hi @marksutherland 

That's great thanks. I have been to TMU but it was quite a long time ago.
Would it be OK to let you know tomorrow which date is going to work best?

Thanks, Tim

**Comment by marksutherland** [comment]
Hi @timtroup , that's fine, let me know when you're sure!

**Comment by timtroup** [comment]
Hi @marksutherland apologies for the slow response. I have just been offered a developer contract in Norway commencing in 3 weeks time so unfortunately I'm not going to have the time to fit a talk in at the moment. Once that concludes I will be back in touch. If anyone if going to JavaZone in Oslo this year then it would be great to meet up.
